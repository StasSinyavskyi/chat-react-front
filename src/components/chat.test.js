/* eslint-env jest */
import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import Chat from './chat';

jest.mock('./chat-messages', () => () => 'ChatMessages');
jest.mock('./message-input', () => () => 'MessageInput');

// if we have several test we can join them in describe
describe('<Chat />', () => {
  const testParam = (
    <Chat
      sendMessage={() => 'sendMessage'}
      joinChat={() => 'joinChat'}
      match={{ params: { chatId: '123' } }}
      activeChat={{ _id: '123' }}
      messages={{ }}
      activeUser={{ isMember: true, isCreator: true, isChatMember: true }}
      isConnected={false}
    />
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(testParam, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render correctly', () => {
    const tree = renderer
      .create(testParam)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
