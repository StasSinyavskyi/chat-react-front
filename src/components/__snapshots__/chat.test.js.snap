// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> render correctly 1`] = `
<main
  className="Chat-chatLayout-2"
>
  ChatMessages
  MessageInput
</main>
`;
